.github/workflows/ci.yaml

```diff
@@ -25,7 +25,7 @@ jobs:
       - name: setup-go
         uses: actions/setup-go@v5
         with:
-          go-version: 1.21.x
+          go-version: 1.22.x
```

Comment:
If `go.mod` states Go 1.20, and we're not using a matrix here to run these steps across multiple versions, seems like we should actually set this to `1.20.x` instead of updating it to 1.22. Why should this be using latest?

Reply:
Was following current, suppose its better to test old versions of go then latest. Have set at 1.20.x.

Reply:
Reverted, lint breaks and is sensitive to go version. We could create a matrix for these test cases but I think it's safe for example code to be run at the latest version.

Reply:
> safe for example code to be run at the latest version

Then we should change `go.mod` to 1.22, or else we're not actually verifying that the repo works, like for people that pull down the contents and use an earlier supported version.

I think it's better to just leave this at the older version. Or we could move it to Go 1.21, since that is the oldest Go version still getting updates.

> lint breaks and is sensitive to go version

Shouldn't it be sensitive to the version of golangci-lint, not the version of go? Or if there's a breakage due to gofmt changes from 1.20 to 1.22, we can just format it with 1.20 for now.